### ISSUE TITLE
Viewers

### ISSUE THREAD

**Issue by ljcollins25** [Ref12/Codex#47]
The idea behind this feature is to provide a rich viewers for indexed files or files which are currently not indexed. The core thought is to have the view defined in XAML which is databound to object model defined in JSON. For instance, dlls checked into the repo might have metadata extracted into json and displayed in an tree view.

**Comment by KirillOsenkov** [comment]
or alternatively just store a text which is a human readable representation of file content, such as ildasm dump for dll files or hex dump for unknown binaries. Advantages are that these are diffable.

**Comment by KirillOsenkov** [comment]
of course we should ideally have both

**Comment by ljcollins25** [comment]
Agreed, I'm thinking their will be a "View Code" option for seeing the raw text underlying the view which may or may not be the same as the json. The json file would be diffable as well, but depending on what data is in the json file it might or might not be easy to read.
